### DISCUSSION TITLE
Automatic power switch power tracking

### DISCUSSION THREAD

**Discussion by yangsystem** [netbox-community/netbox#17757] [Help Wanted!]
I just discovered that the ATS power utilization is not shown on the rack page.

I have one "Power panel" witch has 2 "Power feeds".
I have a rack with 2 PDU that link directly to the "Power feeds".

I connect one server: PSU1 to PDU1 port1 and PSU2 to PDU2 port1, the server has 1000W "Max draw" and 1000W "Allocated draw".
The "Power Utilization" bar will show me that I have a load.

If I put in the same rack an automatic transfer switch (ATS) and link "Power port A" to PDU1 port 5 and "Power port B" to PDU2 port5 with 0W  "Max draw" "Allocated draw" (ATS supposedly do not consume power).
Then I connect to ATS port1 a workstation (1000W "Max draw" and 1000W "Allocated draw").

Now the "Power Utilization" bar will show me only the load for the server not the load allocated also for the workstation.

How to resolve this problem?

**Comment by candlerb** [comment]
The power calculation features in Netbox are not well documented. I attempted to reverse-engineer them in discussion #12837. I don't think anything has changed since then.

For your particular issue: Netbox does not sum up power port utilization recursively. The "allocated" figure on a power feed is the sum  of "allocated draw" ports one hop away, but does not include grandchildren two hops away. That's just how it is.

You could manually set the allocated draw on the ATS power port to the sum of its children, but that's a pain to maintain.

TBH, I think the power calculation features in Netbox are not particularly useful in their current form. If you want to do anything more sophisticated, you can write custom scripts to aggregate power usage in whatever way you like, such as [this one](https://github.com/netbox-community/customizations/blob/master/scripts/power_summary.py).
